microsoft/testsuites/core/storage.py

```diff
@@ -270,6 +282,28 @@ def verify_scsi_disk_controller_type(self, node: RemoteNode) -> None:
     def verify_nvme_disk_controller_type(self, node: RemoteNode) -> None:
         self._verify_disk_controller_type(node, DiskControllerType.NVME)
 
+    @TestCaseMetadata(
+        description="""
+        This test verifies the nvme drivers are either built in or present in initramfs.
+        This test can be used to verify an image before it is tagged as NVMe capable.
+        """,
+        priority=1,
+    )
+    def verify_supports_nvme(self, node: RemoteNode) -> None:
+        _has_nvme_core = node.tools[KernelConfig].is_built_in("CONFIG_NVME_CORE") or (
+            node.tools[KernelConfig].is_built_as_module("CONFIG_NVME_CORE")
+            and node.tools[Lsinitrd].has_module("nvme-core.ko")
+        )
+        _has_nvme = node.tools[KernelConfig].is_built_in("CONFIG_BLK_DEV_NVME") or (
+            node.tools[KernelConfig].is_built_as_module("CONFIG_BLK_DEV_NVME")
+            and node.tools[Lsinitrd].has_module("nvme.ko")
+        )
+        if not (_has_nvme_core and _has_nvme):
```

Comment:
Where is the validation line? When will the test case fail?

Reply:
This test case only checks NVMe capability for the image. You asked me to skip the test is the image is not NVMe capable. What would you like the behavior to be?

Reply:
As discussed offline, the logic should be like,
expected: False, actual: False -> Skipped
expected: False, actual: True -> Skipped (because this value is not a concern in this scenario)
expected: True, actual: False -> Failed
expected: True, actual: True -> Passed

In most cases, the expected values can be obtained from the test environments by vm_size or other information. If it's not, you can pass in a test variable. The default expected value should be False to prevent the test case fails, if the tag is not cared.